File: packages/cli-python/src/spec_init/checklist.py
"""Post-init checklist: the "what to do next" summary after scaffolding."""

from __future__ import annotations

from collections.abc import Sequence

from .args import Integration, Persona
from .integrations import INTEGRATION_REGISTRY, InstallResult

PERSONA_GUIDANCE: dict[Persona, str] = {
    "vibe": "Ship first. Ask Claude to explain any heavy step in one line before doing it - but never skip a required cross-question.",
    "student": 'After each phase, ask Claude "Why this step?" and append the answer to claude/learnings.md.',
    "engineer": "Read claude/context.md before Grep (cheaper), and log every non-trivial decision to claude/design-decisions.md.",
    "team": "Every PR runs the checklist in .github/pull_request_template.md; treat unresolved memory-file conflicts as blocking.",
}


def post_init_checklist(
    out_dir: str,
    persona: Persona | None,
    integrations: Sequence[Integration] = (),
    install_results: Sequence[InstallResult] = (),
    verb: str = "Scaffolded",
) -> str:
    """Return the post-init checklist as a string. `out_dir` is the scaffold path."""
    lines = ["", f"[OK] {verb} {out_dir}", ""]

    if persona:
        lines += [
            f"Persona: {persona}",
            f"  {PERSONA_GUIDANCE[persona]}",
            "  Change it any time with: spec-init customize --persona <name>",
            "",
        ]

    lines += [
        "Next steps:",
        f"  1. cd {out_dir}",
        "  2. Open the project in Claude Code",
        "  3. Run /spec-requirements to build claude/srs.md",
        "  4. Then /spec-design and /spec-plan",
        "",
    ]

    if integrations:
        lines.append("Integrations enabled:")
        for name in integrations:
            info = INTEGRATION_REGISTRY[name]
            res = next((r for r in install_results if r.name == name), None)
            tag = ""
            if res and res.status == "installed":
                tag = "  [installed]"
            elif res and res.status == "failed":
                tag = f"  [install failed - run: {res.command}]"
            elif res and res.status == "manual":
                tag = f"  [install: {res.command}]"
            lines.append(f"  - {info.title}{tag}")
            lines.append(f"    docs: {info.docs_url}")
        lines.append("")

    return "\n".join(lines)

File: packages/cli-node/src/checklist.ts
// Post-init checklist: the "what to do next" summary after scaffolding (FR-INIT-04).

import type { Integration, Persona } from './args.js';
import { INTEGRATION_REGISTRY } from './integrations.js';
import type { InstallResult } from './integrations.js';

/** One-line, persona-specific guidance printed after a scaffold. */
export const PERSONA_GUIDANCE: Record<Persona, string> = {
  vibe: 'Ship first. Ask Claude to explain any heavy step in one line before doing it - but never skip a required cross-question.',
  student: 'After each phase, ask Claude "Why this step?" and append the answer to claude/learnings.md.',
  engineer: 'Read claude/context.md before Grep (cheaper), and log every non-trivial decision to claude/design-decisions.md.',
  team: 'Every PR runs the checklist in .github/pull_request_template.md; treat unresolved memory-file conflicts as blocking.',
};

/** Return the post-init checklist as a string. `outDir` is the scaffold path. */
export function postInitChecklist(
  outDir: string,
  persona: Persona | null,
  integrations: readonly Integration[] = [],
  installResults: readonly InstallResult[] = [],
  verb = 'Scaffolded',
): string {
  const lines = ['', `[OK] ${verb} ${outDir}`, ''];

  if (persona) {
    lines.push(`Persona: ${persona}`);
    lines.push(`  ${PERSONA_GUIDANCE[persona]}`);
    lines.push('  Change it any time with: spec-init customize --persona <name>');
    lines.push('');
  }

  lines.push('Next steps:');
  lines.push(`  1. cd ${outDir}`);
  lines.push('  2. Open the project in Claude Code');
  lines.push('  3. Run /spec-requirements to build claude/srs.md');
  lines.push('  4. Then /spec-design and /spec-plan');
  lines.push('');

  if (integrations.length > 0) {
    lines.push('Integrations enabled:');
    for (const name of integrations) {
      const info = INTEGRATION_REGISTRY[name];
      const res = installResults.find((r) => r.name === name);
      let tag = '';
      if (res?.status === 'installed') tag = '  [installed]';
      else if (res?.status === 'failed') tag = `  [install failed - run: ${res.command}]`;
      else if (res?.status === 'manual') tag = `  [install: ${res.command}]`;
      lines.push(`  - ${info.title}${tag}`);
      lines.push(`    docs: ${info.docsUrl}`);
    }
    lines.push('');
  }

  return lines.join('\n');
}
